core/src/test/scala/org/apache/spark/SparkFunSuite.scala

```diff
@@ -105,5 +105,16 @@ abstract class SparkFunSuite
       logInfo(s"\n\n===== FINISHED $shortSuiteName: '$testName' =====\n")
     }
   }
-
+  /**
+   * Creates a temporary directory, which is then passed to `f` and will be deleted after `f`
+   * returns.
+   *
+   * @todo Probably this method should be moved to a more general place
+   */
+  protected def withCreateTempDir(f: File => Unit): Unit = {
```

Comment:
Is there any reason not to use `withTempDir` as a function name like other modules?

Reply:
`trait SQLTestUtils extends SparkFunSuite with SQLTestUtilsBase with PlanTest`
if `SparkFunSuite `and `SQLTestUtilsBase `use the same name `withTempDir`. 
Can cause name contamination? thanks.

Reply:
if we have this function in `SparkFunSuite`, why do we need to define it again in `SQLTestUtils`?

Reply:
shall we also do `waitForTasksToFinish` in `withCreateTempDir`?

Reply:
I'm not sure if I need `assert(spark.sparkContext.statusTracker      .getExecutorInfos.map(_.numRunningTasks()).sum == 0)`.  after all, `protected def spark: SparkSession`  defined in `SQLTestData`.  Unless we construct one `waitForTasksToFinish ` look like
```
protected def waitForTasksToFinish(): Unit = {
    eventually(timeout(10.seconds)) {
    }
  }
```

Reply:
OK, how about in `SQLTestUtils` we override `withTempDir` with this extra logic?

Reply:
Sorry, maybe I didn't make it clear.  `withTempDir ` define in `SQLTestUtilsBase` not in `SQLTestUtils `.if we will override  `withTempDir `  first must be remove  `withTempDir ` from `SQLTestUtilsBase` to `SQLTestUtils` and override,  but in this way, other functions are also involved.

Reply:
I'm not talking about details like which calss to override, just the idea. Why wouldn't override work?

Reply:
Yes, it seems like we should be able to use an override. The subclass that needs to inject an additional method call in the block can call the super method with a lambda that calls the user-supplied block, then this other method. It's probably worth whatever surgery is needed to make this clean and reduce duplication. We already have a lot of "create temp thing" methods all over.